File: eventual-consistency-support-core/src/main/java/org/xyattic/eventual/consistency/support/core/lock/RedisLock.java
package org.xyattic.eventual.consistency.support.core.lock;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author wangxing
 * @create 2020/4/10
 */
@Target({ElementType.ANNOTATION_TYPE, ElementType.TYPE, ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Inherited
public @interface RedisLock {
}

File: eventual-consistency-support-core/src/main/java/org/xyattic/eventual/consistency/support/core/lock/RedisLock.kt
package org.xyattic.eventual.consistency.support.core.lock

import java.lang.annotation.Inherited

/**
 * @author wangxing
 * @create 2020/4/10
 */
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention
@MustBeDocumented
@Inherited
annotation class RedisLock